tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetFramework.cs

```diff
@@ -9,25 +9,34 @@
 using System.IO;
 using System.Reflection;
 
+#nullable enable
+
 namespace Datadog.Trace.ClrProfiler.Managed.Loader
 {
     /// <summary>
     /// A class that attempts to load the Datadog.Trace .NET assembly.
     /// </summary>
     public partial class Startup
     {
-        private static string ResolveManagedProfilerDirectory()
+        private static string? ResolveManagedProfilerDirectory()
         {
             var tracerHomeDirectory = ReadEnvironmentVariable("DD_DOTNET_TRACER_HOME") ?? string.Empty;
-            return Path.Combine(tracerHomeDirectory, "net461");
+            var fullPath = Path.Combine(tracerHomeDirectory, "net461");
+            if (!Directory.Exists(fullPath))
```

Comment:
Since the goal of this PR is to avoid exceptions that could crash the process, we should probably wrap `Directory.Exists()` in a `try/catch` in case it throws an exception. Possible causes could be an invalid value in `DD_DOTNET_TRACER_HOME` or the lack of ACL permissions.